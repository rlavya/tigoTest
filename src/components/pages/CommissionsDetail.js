import { connect } from "react-redux";
import { siteTree } from "../../siteTree";
import { dataForId, configToId } from "../../util/db";
import { redirectTo } from "../../router";
import TransactionsOrCommissionsDetail from "../components/TransactionsOrCommissionsDetail";

export default connect(
  state => {
    const items = dataForId(
      state,
      configToId("CommissionDetail", "list", true)
    );
    return {
      items,
      pageName: siteTree.TransactionsDetail.name,
      periodFilter: state.route.periodFilter || "hora",
      transactionFilter: state.route.transactionFilter || "todas"
    };
  },
  dispatch => {
    return {
      changePeriodFilter(filter) {
        redirectTo({ periodFilter: filter }, true);
      },
      changeTransactionFilter(filter) {
        redirectTo({ transactionFilter: filter }, true);
      }
    };
  }
)(TransactionsOrCommissionsDetail);
