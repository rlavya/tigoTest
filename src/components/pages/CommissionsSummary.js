import { connect } from "react-redux";
import { siteTree } from "../../siteTree";
import { dataForId, configToId } from "../../util/db";
import TransactionsOrCommissionsSummary from "../components/TransactionsOrCommissionsSummary";

export default connect(state => {
  const commissions = dataForId(state, configToId("Commission", "list", true));

  return {
    items: commissions,
    pageName: siteTree.CommissionsSummary.name,
    detailViewPageName: siteTree.CommissionsDetail.name
  };
})(TransactionsOrCommissionsSummary);
