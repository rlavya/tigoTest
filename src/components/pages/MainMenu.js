import React, { Component } from "react";
import PageTitle from "../components/PageTitle";
import { List, ListItem } from "material-ui/List";
import TextField from "material-ui/TextField";
import ChevronRight from "material-ui/svg-icons/navigation/chevron-right";
import Divider from "material-ui/Divider";
import { redirectTo } from "../../router";
import styled from "styled-components";
import { connect } from "react-redux";
import { update as updateClientPhone } from "../../actionCreators/clientPhone";
import { Card } from "material-ui/Card";
import { lightBlue200, indigo300 } from "material-ui/styles/colors";
import get from "lodash/get";
import { convertMS } from "../../util";
import { clearCurrentUserAction } from "../../actionCreators/selling";
import {
  logoutAction,
  requestBalanceAction,
  toggleBalanceEpinPopup,
  updateBalanceEpin,
  getBalanceAction
} from "../../actionCreators/auth";
import FlatButton from "material-ui/FlatButton";
import Refresh from "material-ui/svg-icons/navigation/refresh.js";
import PinDialog from "../components/PinDialog";

import { track } from "../../util/tracker";

const ClientPhoneNumberWrap = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 20px 0;
  width: 230px;
  margin: 0 auto;
  img {
    margin-right: 5px;
    margin-left: -24px;
    margin-top: 4px;
  }
`;

const ClearButtonWrap = styled.div`
  width: 30px;
  margin-left: -30px;
  margin-bottom: -17px;
`;

const CardWrap = styled.div`
  padding: 15px;
`;

const BalanceWrap = styled.div`
  padding: 5px;
  text-align: right;
  position: relative;
  margin-top: -7px;
`;

const BalanceRequestedTimeWrap = styled.div`
  width: 150px;
  position: absolute;
  top: 45px;
  right: 40px;
  font-weight: 300;
  color: #839ebf;
`;

class BalanceRequestedTime extends Component {
  constructor(props) {
    super(props);
    this.state = { now: Date.now() };
  }
  tick() {
    this.setState({ now: Date.now() });
  }
  componentDidMount() {
    this.interval = setInterval(this.tick.bind(this), 1000);
  }
  componentWillUnmount() {
    clearInterval(this.interval);
  }
  formatHours(h) {
    if (h === 1) {
      return h + " hora";
    }
    return h + " horas";
  }
  formatMinutes(m) {
    if (m === 1) {
      return m + " min";
    }
    return m + " mins";
  }
  render() {
    if (!this.props.receivedAt) {
      return null;
    }
    const diff = convertMS(this.state.now - this.props.receivedAt);
    let diffMessage;

    if (diff.h && diff.h > 0) {
      diffMessage = this.formatHours(Math.max(diff.h, 0));
    } else {
      diffMessage = this.formatMinutes(Math.max(diff.m, 0));
    }

    return (
      <BalanceRequestedTimeWrap className="text-small">
        Actualizado hace {diffMessage}
      </BalanceRequestedTimeWrap>
    );
  }
}

class Balance extends Component {
  renderBalance() {
    if (!this.props.balance) {
      return null;
    }
    return (
      <div className="text-white text-big">
        {this.props.balance.currency}
        {this.props.balance.balance}
      </div>
    );
  }
  render() {
    return (
      <BalanceWrap>
        <div className="left">
          <div className="text-white text-small">Tu Saldo</div>
          {this.renderBalance()}
        </div>
        <Refresh
          onClick={this.props.refresh}
          className="right mt1 pointer"
          color="#5CCAF9"
          style={{ width: "35px", height: "35px" }}
        />
        <BalanceRequestedTime
          receivedAt={get(this.props, "balance.receivedAt")}
        />
      </BalanceWrap>
    );
  }
}

const ConnectedBalance = connect(
  state => {
    return {
      balance: get(state, "auth.balance")
    };
  },
  dispatch => {
    return {
      refresh: () => dispatch(requestBalanceAction())
    };
  }
)(Balance);

export const ConnectedEpinDialog = connect(
  state => {
    return {
      open: !!get(state, "auth.balanceEpinOpen"),
      pin: get(state, "auth.pin") || ""
    };
  },
  dispatch => {
    return {
      onPinChange: (e, value) => dispatch(updateBalanceEpin(value)),
      handleClose: () => dispatch(toggleBalanceEpinPopup(false)),
      handleMainAction: () => {
        dispatch(toggleBalanceEpinPopup(false));
        dispatch(getBalanceAction());
      }
    };
  }
)(PinDialog);

class ClientPhoneNumber extends Component {
  render() {
    return (
      <ClientPhoneNumberWrap>
        <img alt="" src="/images/icn-ClientPhoneNumber-bigger.png" />
        <TextField
          id="client-phone"
          floatingLabelText="Ingresa el número telefónico del cliente:"
          floatingLabelFixed={true}
          onChange={this.props.onInputChange}
          value={this.props.clientPhoneNumber}
          underlineFocusStyle={{ borderColor: lightBlue200 }}
          floatingLabelStyle={{ color: indigo300, width: "320px" }}
          inputStyle={{ fontSize: "1.5rem" }}
          type="tel"
          pattern="[0-9]*"
        />
        <ClearButtonWrap>
          {this.props.clientPhoneNumber !== "" ? (
            <FlatButton
              secondary={true}
              label="×"
              onClick={this.props.clear}
              fullWidth={true}
              labelStyle={{ fontSize: "24px" }}
            />
          ) : null}
        </ClearButtonWrap>
      </ClientPhoneNumberWrap>
    );
  }
}

const ConnectedClientPhoneNumber = connect(
  state => {
    const clientPhoneNumber = get(state, "clientPhone.phoneNumber") || "";

    return {
      clientPhoneNumber
    };
  },
  dispatch => {
    return {
      onInputChange: (e, value) => {
        dispatch(updateClientPhone(value));
      },
      clear: () => dispatch(clearCurrentUserAction())
    };
  }
)(ClientPhoneNumber);

export class MainMenu extends Component {
  render() {
    let listItemStyle = this.props.clientPhone.isValid ? {} : { opacity: 0.5 };
    return (
      <PageTitle page="MainMenu" rightElement={<ConnectedBalance />}>
        <div>
          <CardWrap>
            <Card>
              <ConnectedClientPhoneNumber />
              <List>
                <Divider />
                <ListItem
                  leftAvatar={<img src="/images/icn-recharge-on.png" alt="" />}
                  primaryText="Recargas"
                  rightIcon={<ChevronRight color="#4CC5F2" />}
                  onClick={() => {
                    if (this.props.clientPhone.isValid) {
                      redirectTo({ page: "TopupDataEntry" });
                    } else {
                      this.props.trackClickOnDisabled(
                        this.props.msisdn,
                        this.props.dmsid
                      );
                    }
                  }}
                  style={listItemStyle}
                />
                <Divider />
                <ListItem
                  leftAvatar={<img src="/images/icn-packages-on.png" alt="" />}
                  primaryText="Paquetes"
                  rightIcon={<ChevronRight color="#4CC5F2" />}
                  onClick={() => {
                    if (this.props.clientPhone.isValid) {
                      redirectTo({ page: "PackagesCategories" });
                    } else {
                      this.props.trackClickOnDisabled(
                        this.props.msisdn,
                        this.props.dmsid
                      );
                    }
                  }}
                  style={listItemStyle}
                />
              </List>
            </Card>
          </CardWrap>
          <CardWrap>
            <Card>
              <List>
                <ListItem
                  primaryText="Ventas"
                  leftAvatar={<img src="/images/icn-sales.png" alt="" />}
                  rightIcon={<ChevronRight color="#4CC5F2" />}
                  onClick={() => {
                    redirectTo({ page: "SalesSummary" });
                  }}
                />
                <Divider />
                <ListItem
                  primaryText="Cambio de PIN"
                  leftAvatar={<img src="/images/icn-ChangePin.png" alt="" />}
                  rightIcon={<ChevronRight color="#4CC5F2" />}
                  onClick={() => {
                    redirectTo({ page: "ChangePin" });
                  }}
                />
                <Divider />
                <ListItem
                  leftAvatar={<img src="/images/icn-logout.png" alt="" />}
                  primaryText="Salir"
                  rightIcon={<ChevronRight color="#4CC5F2" />}
                  onClick={this.props.logout}
                />
              </List>
            </Card>
          </CardWrap>

          <ConnectedEpinDialog />
        </div>
      </PageTitle>
    );
  }
}

export default connect(
  state => {
    const msisdn = get(state, "auth.login.msisdn");
    const dmsid = get(state, "dmsid");
    const clientPhone = get(state, "clientPhone");
    return {
      clientPhone,
      dmsid,
      msisdn
    };
  },
  dispatch => {
    return {
      logout: () => dispatch(logoutAction()),
      trackClickOnDisabled: (posMsisdn, posDmsid) => {
        const event =
          "Home screen No Tigo MSISDN Recharge/Package sell attempt";
        navigator.geolocation.getCurrentPosition(
          info => {
            track(event, {
              posMsisdn,
              posDmsid,
              latitude: get(info, "coords.latitude"),
              longitude: get(info, "coords.longitude")
            });
          },
          () => {
            track(event, { posMsisdn, posDmsid });
          },
          { maximumAge: 300000 }
        );
      }
    };
  }
)(MainMenu);
