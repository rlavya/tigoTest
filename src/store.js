import { applyMiddleware, createStore } from "redux";
import { createLogger } from "redux-logger";
import { composeWithDevTools } from "redux-devtools-extension/developmentOnly";
import thunk from "redux-thunk";
import reducer from "./reducer";

const getMiddleware = () => {
  return applyMiddleware(createLogger(), thunk);
};

const store = createStore(reducer, composeWithDevTools(getMiddleware()));

export default store;
