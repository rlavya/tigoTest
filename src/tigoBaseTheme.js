const tigoBaseTheme = {
  fontFamily: "Roboto, sans-serif",
  borderRadius: 2,
  palette: {
    primary1Color: "#00377B",
    primary2Color: "#4CC5F2",
    primary3Color: "#333",
    accent1Color: "#4CC5F2",
    accent2Color: "#4CC5F2",
    accent3Color: "#4CC5F2",
    textColor: "#00377B",
    secondaryTextColor: "#839ebf",
    alternateTextColor: "#fafafa",
    canvasColor: "#FFFFFF"
  }
};

export default tigoBaseTheme;
