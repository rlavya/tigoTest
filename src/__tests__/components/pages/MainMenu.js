import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import getMuiTheme from "material-ui/styles/getMuiTheme";
import tigoBaseTheme from "../../../tigoBaseTheme";
import { MainMenu } from "../../../components/pages/MainMenu";
import renderer from "react-test-renderer";
import React from "react";
import store from "../../../store";
import {Provider} from "react-redux";

it("renders correctly", () => {
  const rendered = renderer.create(
    <MuiThemeProvider muiTheme={getMuiTheme(tigoBaseTheme)}>
      <Provider store={store}>
        <MainMenu clientPhone={{ isValid: true }} />
      </Provider>
    </MuiThemeProvider>
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});
