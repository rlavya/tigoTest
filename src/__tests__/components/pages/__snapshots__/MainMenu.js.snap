// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <div
      className={undefined}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#00377B",
          "borderRadius": "0px",
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "#00377B",
          "display": "flex",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "paddingLeft": 24,
          "paddingRight": 24,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": "100%",
          "zIndex": 1100,
        }
      }
    >
      <div
        style={
          Object {
            "marginLeft": -16,
            "marginRight": 8,
            "marginTop": 8,
            "muiPrepared": true,
          }
        }
      >
        <span />
      </div>
      <div
        onClick={[Function]}
        style={
          Object {
            "boxFlex": 1,
            "color": "#fafafa",
            "flex": "1",
            "fontSize": 24,
            "fontWeight": 400,
            "height": 64,
            "letterSpacing": 0,
            "lineHeight": "64px",
            "margin": 0,
            "muiPrepared": true,
            "overflow": "hidden",
            "paddingTop": 0,
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
          }
        }
      >
        <span
          style={
            Object {
              "display": "inline-block",
              "paddingTop": "11px",
            }
          }
        >
          <img
            alt="TiGo PoS"
            src="/images/logo-TigoPOS.png"
          />
        </span>
      </div>
      <div
        style={
          Object {
            "marginLeft": "auto",
            "marginRight": -16,
            "marginTop": 8,
            "muiPrepared": true,
          }
        }
      >
        <div
          className="sc-bxivhb jqGatY"
        >
          <div
            className="left"
          >
            <div
              className="text-white text-small"
            >
              SALDO ePIN
            </div>
          </div>
          <svg
            className="right mt1 pointer"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#00377B",
                "display": "inline-block",
                "fill": "#5CCAF9",
                "height": "35px",
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": "35px",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-htpNat UxadW"
  >
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "#00377B",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <div
          className="sc-bdVaJa ljQZmJ"
        >
          <img
            src="/images/icn-ClientPhoneNumber.png"
          />
          <div
            className={undefined}
            style={
              Object {
                "backgroundColor": "transparent",
                "cursor": "auto",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 16,
                "height": 72,
                "lineHeight": "24px",
                "muiPrepared": true,
                "position": "relative",
                "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": 256,
              }
            }
          >
            <label
              className={undefined}
              htmlFor="client-phone"
              onClick={undefined}
              style={
                Object {
                  "color": "#7986cb",
                  "lineHeight": "22px",
                  "muiPrepared": true,
                  "pointerEvents": "none",
                  "position": "absolute",
                  "top": 38,
                  "transform": "scale(0.75) translate(0, -28px)",
                  "transformOrigin": "left top",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "zIndex": 1,
                }
              }
            >
              Tel√©fono del cliente:
            </label>
            <input
              disabled={false}
              id="client-phone"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "WebkitOpacity": 1,
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "appearance": "textfield",
                  "backgroundColor": "rgba(0,0,0,0)",
                  "border": "none",
                  "boxSizing": "border-box",
                  "color": "#00377B",
                  "cursor": "inherit",
                  "font": "inherit",
                  "height": "100%",
                  "marginTop": 14,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "width": "100%",
                }
              }
              type="text"
              value=""
            />
            <div>
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 2,
                    "borderColor": "#81d4fa",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "transform": "scaleX(0)",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
          <div
            className="sc-bwzfXH jhKOLS"
          >
            <button
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "backgroundColor": "rgba(0, 0, 0, 0)",
                  "border": 10,
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "color": "#4CC5F2",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 36,
                  "lineHeight": "36px",
                  "margin": 0,
                  "minWidth": "100%",
                  "muiPrepared": true,
                  "outline": "none",
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "verticalAlign": null,
                }
              }
              tabIndex={0}
              type="button"
            >
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <span
                  style={
                    Object {
                      "fontSize": "24px",
                      "fontWeight": 500,
                      "letterSpacing": 0,
                      "muiPrepared": true,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                      "position": "relative",
                      "textTransform": "uppercase",
                      "verticalAlign": "middle",
                    }
                  }
                >
                  √ó
                </span>
              </div>
            </button>
          </div>
        </div>
        <div
          style={
            Object {
              "muiPrepared": true,
              "padding": "8px 0px 8px 0px",
            }
          }
        >
          <hr
            style={
              Object {
                "backgroundColor": "#e0e0e0",
                "border": "none",
                "height": 1,
                "margin": 0,
                "marginLeft": 0,
                "marginTop": -1,
                "muiPrepared": true,
              }
            }
          />
          <div>
            <span
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              open={null}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "background": "none",
                  "backgroundColor": null,
                  "border": 10,
                  "boxSizing": "border-box",
                  "color": "#00377B",
                  "cursor": "pointer",
                  "display": "block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": 16,
                  "fontWeight": "inherit",
                  "lineHeight": "16px",
                  "margin": 0,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "verticalAlign": null,
                }
              }
              tabIndex={0}
            >
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  style={
                    Object {
                      "marginLeft": 0,
                      "muiPrepared": true,
                      "paddingBottom": 20,
                      "paddingLeft": 72,
                      "paddingRight": 56,
                      "paddingTop": 20,
                      "position": "relative",
                    }
                  }
                >
                  <svg
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "color": "#00377B",
                        "display": "block",
                        "fill": "#4CC5F2",
                        "height": 24,
                        "margin": 12,
                        "muiPrepared": true,
                        "position": "absolute",
                        "right": 4,
                        "top": 4,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "userSelect": "none",
                        "width": 24,
                      }
                    }
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                  <img
                    alt=""
                    src="/images/icn-recharge-on.png"
                    style={
                      Object {
                        "left": 16,
                        "position": "absolute",
                        "top": 8,
                      }
                    }
                  />
                  <div
                    style={
                      Object {
                        "muiPrepared": true,
                      }
                    }
                  >
                    Recargas
                  </div>
                </div>
              </div>
            </span>
          </div>
          <hr
            style={
              Object {
                "backgroundColor": "#e0e0e0",
                "border": "none",
                "height": 1,
                "margin": 0,
                "marginLeft": 0,
                "marginTop": -1,
                "muiPrepared": true,
              }
            }
          />
          <div>
            <span
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              open={null}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "background": "none",
                  "backgroundColor": null,
                  "border": 10,
                  "boxSizing": "border-box",
                  "color": "#00377B",
                  "cursor": "pointer",
                  "display": "block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": 16,
                  "fontWeight": "inherit",
                  "lineHeight": "16px",
                  "margin": 0,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "verticalAlign": null,
                }
              }
              tabIndex={0}
            >
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  style={
                    Object {
                      "marginLeft": 0,
                      "muiPrepared": true,
                      "paddingBottom": 20,
                      "paddingLeft": 72,
                      "paddingRight": 56,
                      "paddingTop": 20,
                      "position": "relative",
                    }
                  }
                >
                  <svg
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "color": "#00377B",
                        "display": "block",
                        "fill": "#4CC5F2",
                        "height": 24,
                        "margin": 12,
                        "muiPrepared": true,
                        "position": "absolute",
                        "right": 4,
                        "top": 4,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "userSelect": "none",
                        "width": 24,
                      }
                    }
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                  <img
                    alt=""
                    src="/images/icn-packages-on.png"
                    style={
                      Object {
                        "left": 16,
                        "position": "absolute",
                        "top": 8,
                      }
                    }
                  />
                  <div
                    style={
                      Object {
                        "muiPrepared": true,
                      }
                    }
                  >
                    Paquetes
                  </div>
                </div>
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-htpNat UxadW"
  >
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "#00377B",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <div
          style={
            Object {
              "muiPrepared": true,
              "padding": "8px 0px 8px 0px",
            }
          }
        >
          <div>
            <span
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              open={null}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "background": "none",
                  "backgroundColor": null,
                  "border": 10,
                  "boxSizing": "border-box",
                  "color": "#00377B",
                  "cursor": "pointer",
                  "display": "block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": 16,
                  "fontWeight": "inherit",
                  "lineHeight": "16px",
                  "margin": 0,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "verticalAlign": null,
                }
              }
              tabIndex={0}
            >
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  style={
                    Object {
                      "marginLeft": 0,
                      "muiPrepared": true,
                      "paddingBottom": 16,
                      "paddingLeft": 16,
                      "paddingRight": 56,
                      "paddingTop": 16,
                      "position": "relative",
                    }
                  }
                >
                  <svg
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "color": "#00377B",
                        "display": "block",
                        "fill": "#4CC5F2",
                        "height": 24,
                        "margin": 12,
                        "muiPrepared": true,
                        "position": "absolute",
                        "right": 4,
                        "top": 0,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "userSelect": "none",
                        "width": 24,
                      }
                    }
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                  <div
                    style={
                      Object {
                        "muiPrepared": true,
                      }
                    }
                  >
                    Cambiar contrase√±a
                  </div>
                </div>
              </div>
            </span>
          </div>
          <hr
            style={
              Object {
                "backgroundColor": "#e0e0e0",
                "border": "none",
                "height": 1,
                "margin": 0,
                "marginLeft": 0,
                "marginTop": -1,
                "muiPrepared": true,
              }
            }
          />
          <div>
            <span
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              open={null}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "background": "none",
                  "backgroundColor": null,
                  "border": 10,
                  "boxSizing": "border-box",
                  "color": "#00377B",
                  "cursor": "pointer",
                  "display": "block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": 16,
                  "fontWeight": "inherit",
                  "lineHeight": "16px",
                  "margin": 0,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "verticalAlign": null,
                }
              }
              tabIndex={0}
            >
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  style={
                    Object {
                      "marginLeft": 0,
                      "muiPrepared": true,
                      "paddingBottom": 20,
                      "paddingLeft": 72,
                      "paddingRight": 56,
                      "paddingTop": 20,
                      "position": "relative",
                    }
                  }
                >
                  <svg
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "color": "#00377B",
                        "display": "block",
                        "fill": "#4CC5F2",
                        "height": 24,
                        "margin": 12,
                        "muiPrepared": true,
                        "position": "absolute",
                        "right": 4,
                        "top": 4,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "userSelect": "none",
                        "width": 24,
                      }
                    }
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                  <img
                    alt=""
                    src="/images/icn-logout.png"
                    style={
                      Object {
                        "left": 16,
                        "position": "absolute",
                        "top": 8,
                      }
                    }
                  />
                  <div
                    style={
                      Object {
                        "muiPrepared": true,
                      }
                    }
                  >
                    Salir
                  </div>
                </div>
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
